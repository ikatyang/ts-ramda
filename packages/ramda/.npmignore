node_modules/
scripts/
src/
templates/
.npmignore
test.ts
tsconfig.json
tslint.json
