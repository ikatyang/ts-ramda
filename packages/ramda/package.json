{
  "name": "@ikatyang/types-ramda",
  "version": "0.23.0",
  "description": "TypeScript's type definitions for Ramda",
  "repository": "https://github.com/ikatyang/types",
  "author": "ikatyang",
  "license": "MIT",
  "scripts": {
    "ts-node": "../../node_modules/.bin/ts-node -P ./tsconfig.json",
    "build": "yarn run ts-node -- ./scripts/build.ts ./templates ./typings"
  },
  "devDependencies": {
    "ramda": "0.23.0"
  }
}
