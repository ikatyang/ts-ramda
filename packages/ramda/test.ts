/// <reference path="./index.d.ts" />

import * as R from 'ramda';

R.__;

R.add(1, 2);
R.add(1)(2);
