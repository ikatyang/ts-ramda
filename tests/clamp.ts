import * as R from 'ramda';

// @dts-jest:skip number
R.clamp(1, 10, -1); // => 1
// @dts-jest
R.clamp(1, 10)(11); // => 10
// @dts-jest
R.clamp(1)(10, 4);  // => 4
// @dts-jest:skip string
R.clamp('a', 'd', 'e');  // => 'd'
