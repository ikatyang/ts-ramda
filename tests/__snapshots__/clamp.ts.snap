// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`R.clamp(1)(10, 4) 1`] = `"number"`;

exports[`R.clamp(1, 10)(11) 1`] = `"number"`;
